core/src/test/java/org/apache/calcite/test/ReflectiveSchemaTest.java

```diff
@@ -806,6 +806,22 @@ private static boolean isNumeric(Class type) {
         .returnsUnordered("V=1970-01-01");
   }
 
+  @Test public void testDateCanCompare2() {
+    final String sql = "select a.v1, a.v2\n"
```

Comment:
Can we merge `DateCanCompare2` into `DateCanCompare` to test three data types ? Or can you split `testDateCanCompare2` into `testTimeCanCompare` and `testTimestampCanCompare` ?

Reply:
Yes, we can. I have merged them together.